### ISSUE TITLE
Core override changes to inputs do not work properly

### ISSUE THREAD

**Issue by prataczak** [libretro/RetroArch#6641]
## Description

If I want to change hotkeys for a specific core (to disable fast forward for example, if I have it on the R trigger), or, if I would like to turn off the 'turbo' hold button that is built into the main input for each player, I cannot do this from the GUI and have it save properly. However, I can MANUALLY enter in the appropriate code into a core override file, and it will work - but if I then try to save more core overrides at a future date, all of the manual entries I entered in pertaining to inputs are automatically erased.

### Expected behavior

Any changes to hotkeys or inputs are saved as overrides via the GUI, and input related overrides are not erased every time a new override is added.

### Steps to reproduce the bug

1. Try to make changes to hotkeys or input settings for Player 1, and save a core override - settings will not save
2. Manually enter in the appropriate code into a core override file, e.g. the line "input_player1_turbo_axis = "+5"" for NEStopia.cfg
3.  Save an override, and all of the data you manually entered in step 2 is gone.

### Version/Commit
RetroArch 1.7.2

### Environment information

- OS: Windows

**Comment by fraf** [comment]
Still actual in retroarch 1.8.5
The feature to map "Hotkey binds" and "port binds" in "settings->input" menu to any core differently is essential ! Each core has its own original gamepad until xbox/ps2 generation where controllers started to be a "norm".

**Comment by luluco250** [comment]
Can confirm for RetroArch 1.8.4 on Switch.
While the hotkeys seem to be forgotten, the menu combo is remembered.
In my case I have L3 + R3 setup for toggling the menu, L3 alone for rewind and R3 alone for fast-forward toggle.
For GBA games I like to change the combination, since it has more free buttons to use.
I tried making a game override for Sonic Advance where I removed the menu combo, setting L3 as the hotkey for menu toggle, R3 for slow motion toggle, L2 for rewind and R2 for fast forward hold.
It all works fine until I save the game override, then it reverts back to the global config while keeping the menu combo override, thus I become locked out of the menu.
